### ISSUE TITLE
MakePath function in helpers/path.go removes unicode "mark"s

### ISSUE THREAD

**Issue by Brahmasmi** [spf13/hugo#1488]
**Issue**
Hugo picks up incorrect section names for non Latin script sections. This results in hugo using incorrect section names and all the consequent issues.

**Bug**
The MakePath function (https://github.com/spf13/hugo/blob/master/helpers/path.go#L77) removes unicode "mark"s, which changes the meaning of the string. As an example, consider the following program (the snippet assumes that you have hugo source on your machine).
```
package main

import (
        "fmt"

        "github.com/spf13/hugo/helpers"
)

func main() {
        fmt.Println(helpers.MakePath("संस्कृत"))
}
```
The expected output is "संस्कृत" i.e., the string should remain unmodified. However, the actual output is "ससकत", which is a totally different word.

In case you are facing difficulties in viewing those glyphs, please copy the following strings and paste at http://www.google.com/intl/sa/inputtools/try/:
```
संस्कृत
ससकत
```
The first string is the input, as well as the expected output. The second string is what is the current erroneous output.

**Suggested Fix**
On https://github.com/spf13/hugo/blob/master/helpers/path.go#L99,
```
if unicode.IsLetter(r) || unicode.IsDigit(r) || r == '.' || r == '/' || r == '\\' || r == '_' || r == '-' || r == '#' {
```
we probably need to add another check for unicode "mark" - https://golang.org/pkg/unicode/#IsMark

**Comment by bep** [comment]
Just tested by adding a check for mark. Didn't fix it.

**Comment by Brahmasmi** [comment]
Could you please help me with the following:
1) Is the "RemovePathAccents" flag set to true? For correct output, it should be set to false i.e. it should not remove accents from the string.
2) Can you please share the
a) the change and
b) the output that you see for the following program before and after applying the change?
Program:
```
package main

import (
        "fmt"

        "github.com/spf13/hugo/helpers"
)

func main() {
        input := "संस्कृत"
        output := helpers.MakePath(input)
        if output == input {
                fmt.Println("Success")
        } else {
                fmt.Println("Failure - input " + input + ", output " + output)
        }
}
```
Locally, I have tried the following change on the current hugo master
```
if unicode.IsLetter(r) || unicode.IsDigit(r) || unicode.IsMark(r) || r == '.' || r == '/' || r == '\\' || r == '_' || r == '-' || r == '#' {
```
and I see following output for the above program. I assume we are running go version 1.5.1.
Before change
```
Failure - input संस्कृत, output ससकत
```
After change
```
Success
```
In case you are facing difficulties in viewing the output glyphs, please copy the glyphs and paste at http://www.google.com/intl/sa/inputtools/try/

**Comment by lolindrath** [comment]
@Brahmasmi There was a change made to fix #1180 on June 16th, 2015 that seems related. Are you up to date?

When I change your sample to include the RemovePathAccents it seems to work properly:

```
package main

import (
	"fmt"

	"github.com/spf13/hugo/helpers"
	"github.com/spf13/viper"
)

func main() {
	path := "संस्कृत"
	viper.Set("RemovePathAccents", true)
	fmt.Println("Original: ", path)
	fmt.Println("MakePath: ", helpers.MakePath(path))
}
```
